### PR TITLE
feat(cli): `download` command for downloading app sources

### PR DESCRIPTION
**Pull request by jfeodor**

### CODE REVIEW

File: internal/app/current_app_version_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		resp := test.JSONResponse(respBody)
+		doer.On("Do", mock.Anything).Return(resp, nil)
+
+		input := CurrentAppVersionInput{
+			OrganizationSlug: "organization-slug",
+			AppSlug:          "app-slug",
+		}
+		output, err := s.CurrentAppVersion(context.TODO(), input)
+
+		assert.ErrorIs(t, err, ErrAccesDenied)
```
Comment (moroderNumerous) [inline internal/app/current_app_version_test.go:130]: TODO: fix typo
`ErrAccesDenied` --> `ErrAccessDenied`

File: internal/manifest/manifest.go [hunk 2]
```diff
@@ -49,6 +49,9 @@ func LoadManifest(filePath string) (*Manifest, error) {
 	var m Manifest
 
 	if _, err := toml.DecodeFile(filePath, &m); err != nil {
+		// TODO: consider which error to return when loading of deprecated
```
Comment (moroderNumerous) [inline internal/manifest/manifest.go:52]: TODO Before or after release of feature?

File: cmd/download/download_test.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		apps.AssertExpectations(t)
+	})
+
+	t.Run("it does not confirm overwriting when app dir does not exist", func(t *testing.T) {
+		appDir := t.TempDir() + "/some-app-dir"
+		client, downloadURL := newTestHTTPClientWithDownload(t, "streamlit_app.tar")
+		apps := &mockAppService{}
+		apps.On("CurrentAppVersion", mock.Anything, app.CurrentAppVersionInput{OrganizationSlug: orgSlug, AppSlug: appSlug}).Return(app.CurrentAppVersionOutput{AppVersionID: appVersionID}, nil)
+		apps.On("AppVersionDownloadURL", mock.Anything, app.AppVersionDownloadURLInput{AppVersionID: appVersionID}).Return(app.AppVersionDownloadURLOutput{DownloadURL: downloadURL}, nil)
+		confirmCalled := false
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/download/download_test.go:61]: Then I would have to refactor the code to take in an interface instead, which would be more complicated IMO.

File: internal/manifest/manifest.go (hunk 2 above)
Comment (jfeodor) [reply internal/manifest/manifest.go:52]: I think we can leave this for now. This is not new behaviour, but I just noticed it while testing this feature, and wanted to document it.

File: cmd/download/download_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (moroderNumerous) [reply cmd/download/download_test.go:61]: Fair enough!

File: internal/app/current_app_version_test.go (hunk 1 above)
Comment (moroderNumerous) [reply internal/app/current_app_version_test.go:130]: This is not necessarily this commit, I would be ok if we just add a TODO comment in this case 😄

File: cmd/download/download_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+		appDir := t.TempDir() + "/some-app-dir"
+		client, downloadURL := newTestHTTPClientWithDownload(t, "streamlit_app.tar")
+		apps := &mockAppService{}
+		apps.On("CurrentAppVersion", mock.Anything, app.CurrentAppVersionInput{OrganizationSlug: orgSlug, AppSlug: appSlug}).Return(app.CurrentAppVersionOutput{AppVersionID: appVersionID}, nil)
+		apps.On("AppVersionDownloadURL", mock.Anything, app.AppVersionDownloadURLInput{AppVersionID: appVersionID}).Return(app.AppVersionDownloadURLOutput{DownloadURL: downloadURL}, nil)
+
+		err := Download(context.TODO(), client, apps, Input{AppDir: appDir, AppSlug: appSlug, OrgSlug: orgSlug}, confirmAlways)
+
+		assert.NoError(t, err)
+		if assert.DirExists(t, appDir) {
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/download/download_test.go:46]: 6555f260e0b0697bdd53035a2f9b90a2b16cb814

File: internal/app/current_app_version_test.go (hunk 1 above)
Comment (jfeodor) [reply internal/app/current_app_version_test.go]: de4f50af91036d7367d179431c4725863ffce2eb

File: cmd/download/download.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+func downloadArchive(client *http.Client, appDir string, url string) error {
+	resp, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil // TODO: handle this?
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/download/download.go]: 164baf1e5505b8fc2559ac17b1d67d9e3db1b0f2

File: cmd/download/cmd.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	"numerous.com/cli/internal/gql"
+)
+
+var long string = `Download app sources of the specified app.
+
+Downloads the sources of an app into the specified [app directory], or the
+directory named after the app slug.
+
+If an app already exists in the [app directory] and it has a default deployment
+specified organization and app slugs are not required. In this case the app
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/download/cmd.go]: 5931eff3d8ad21b5a5af5a379498a3abb0676a0a

**Review by moroderNumerous — APPROVED** [review]

**Pull request merged**